src/check_result.jl

```diff
@@ -77,16 +81,29 @@ function check_equal(
     @test ActualPrimal === ExpectedPrimal
 end
 
+
+# Some structual differential and a natural differential
+function check_equal(actual::Composite{P, T}, expected; kwargs...) where {T, P}
+    if _can_pass_early(actual, expected)
+        @test true
+    else
+        @assert (T <: NamedTuple)  # it should be a structual differential if we hit this
```

Comment:
Is there a reason why this couldn't be a Tuple? I think it is causing https://github.com/JuliaDiff/ChainRulesCore.jl/issues/266